src/encoded/static/components/filegallery.js

```diff
@@ -169,12 +170,17 @@ export class FileTable extends React.Component {
                 return 'proc';
             });
 
+            // showReplicateNumber matches with show-functionality. It has to
+            // be NOT (!)-ed to match with hide-functionality
+            FileTable.procTableColumns.biological_replicates.hide = () => !showReplicateNumber;
```

Comment:
Would it be better to put this into the FileTable.procTableColumns definition (around line 317) instead of on its own here, or is there a reason this property can only be set if `datasetFiles.length` isn’t zero?

Reply:
That makes sense but is hard. FileTable.procTableColumns is an object by itself and has no knowledge of showReplicateNumber. Using "this" (function context) to specific showReplicateNumber gets hairy because "this" changes depending on if you use function or arrow function. Passing in showReplicate in a parameter is hairy too because sorttable does not accept showReplicateNumber and I am trying to do as less surgery as possible. Does this make sense?